src/ThingsDB.ts

```diff
@@ -0,0 +1,157 @@
+import { encode, decode } from "messagepack";
+import EventType from "./EventType";
+const WebSocketClient = require('websocket').w3cwebsocket;
+
+class ThingsDB {
+    private ws: typeof WebSocketClient;
+    private id: number = 1;
+    private readonly uri: string;
+    private pending: { [index: number]: { resolve: (value: any) => void, reject: (value: any) => void }} = {};
+    private listeners: ((type: number, message: any) => void)[] = [];
+
+    constructor(uri: string = 'ws://127.0.0.1:7681') {
+        this.uri = uri;
+    }
+
+    public connect() {
+        return new Promise((resolve, reject) => {
+            this.ws = new WebSocketClient(this.uri);
+
+            this.ws.onopen = (event: Event) => {
+                resolve(true);
+            }
+
+            this.ws.onclose = (event: CloseEvent) => {
+            }
+
+            this.ws.onmessage = (event: MessageEvent) => {
+                const view = new DataView(event.data);
+                const size = view.getUint32(0, true);
+                const id = view.getUint16(4, true);
+                const type = view.getUint8(6);
+                const check = view.getUint8(7);
+
+                if (type !== (255 - check)) {
+                    console.error('Message type check error');
+                    return;
+                }
+
+                let message: any = null;
+                if (size) {
+                    message = decode(event.data.slice(8));
+                }
+
+                if (id in this.pending) {
```

Comment:
Pending should only be checked for types 16, 17, 18 and 19. _(`16..31` is the range reserved for client responses)_
Other types are never a response to a request and therefore the "id" should ignored.